includes/Admin/Admin_AJAX.php

```diff
@@ -89,27 +89,11 @@ public function get_nonce() {
 	 */
 	public function set_up_environment() {
 		// Verify the nonce before continuing.
-		$valid_request = $this->verify_request( filter_input( INPUT_POST, 'nonce', FILTER_SANITIZE_FULL_SPECIAL_CHARS ) );
+		$this->validate_request();
```

Comment:
I had originally approached this that way before but it was recommended to decouple this and handle the response outside of the verify/validate method. (see https://github.com/10up/plugin-check/pull/97#discussion_r1136040211)

Reply:
Moving the logic into reusable functions makes sense, but to @jjgrainger's point we shouldn't couple the new utility methods to send JSON response data. The response should be handled separately by the AJAX callbacks, the new utility functions should rather throw an `Exception` or return `WP_Error` that can then be returned as JSON.